lib/portlayer/network/context.go

```diff
@@ -1045,6 +1047,13 @@ func (c *Context) AddContainer(h *exec.Handle, options *AddContainerOptions) err
 				return err
 			}
 		}
+		// Check that ports are only opened on published network firewall configuration.
+		if len(options.Ports) > 0 && s.TrustLevel() != executor.Published {
```

Comment:
Since "Opened" already has all ports open, it doesn't make sense to supply any additional ports to open. And currently, how George initially described `peers`, it does not have the ability to expose ports. 

The network topology would be like a bunch of containers on a peers network doing work and communicating with each other. Then, when they finish work, they send it to a published network over a bridge, which will be able to communicate to the world.